src/rebar_prv_compile.erl

```diff
@@ -103,3 +107,23 @@ handle_args(State) ->
     {Args, _} = rebar_state:command_parsed_args(State),
     Jobs = proplists:get_value(jobs, Args, ?DEFAULT_JOBS),
     {ok, rebar_state:set(State, jobs, Jobs)}.
+
+copy_app_dirs(AppName, OldAppDir, AppDir) ->
+    case ec_cnv:to_binary(filename:absname(OldAppDir)) =/=
+        ec_cnv:to_binary(filename:absname(AppDir)) of
+        true ->
+            Name = ec_cnv:to_list(AppName),
+            AppFile = filename:join([OldAppDir, "ebin", Name++".app"]),
+            case filelib:is_file(AppFile) of
+                true ->
+                    file:copy(AppFile,
+                              filename:join([AppDir, "ebin", Name++".app"]));
+                false ->
+                    ok
+            end,
+            filelib:ensure_dir(filename:join(AppDir, "dummy")),
+            [file:make_symlink(filename:join(OldAppDir, Dir), filename:join(AppDir, Dir))
+            || Dir <- ["priv", "include"]];
+        false ->
+            ok
+    end.
```

Comment:
`src` (and other potential `src_dirs`) should be copied too as yrl and xrl source is compiled to erlang source in that directory and some tools (cover and xref at least) rely on `src` being adjacent to `ebin` on r15x

is it feasible to copy any non-app or non-meta (`_checkapps`, `_build`, `.rebar`, potentially `_logs` and `_cover` and similar) dir? subsequent runs can just compare modified times of files to decide whether to copy or not as a potential optimization